### PR TITLE
#696 - Fixed the pagination link issue for the topics

### LINKED ISSUE #696: The pagination on Forums > My discussions on the profile page is giving 404 error.
**Describe the bug**
The pagination on Forums > My discussions on the profile page is giving 404 error. It doesn't work for admin, subscribers or even when we use the "View As" feature. I tried resaving the permalinks but didn' help.

**To Reproduce**
1. Go to 'profile > forums' page
2. Click on 'my discussions'
3. Scroll down to 'pagination' and click on the next page

**Expected behavior**
Pagination should work fine. 

**Screenshots**
- https://cl.ly/d8f84ef1904b

**Support ticket links**
- https://secure.helpscout.net/conversation/1125937624/66586/

### PR DESCRIPTION
**Pull request by janakkaneriya007**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #696 .

### How to test the changes in this Pull Request:

1. Go to 'profile > forums' page
2. Click on 'my discussions'
3. Scroll down to 'pagination' and click on the next page

### Proof Screenshots or Video

https://www.loom.com/share/2a097c4578614b49aefcbd74c43b0f8b

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Fixed the pagination link issue for the topics

### CODE REVIEW

File: src/bp-forums/users/template.php [hunk 1]
```diff
@@ -1410,23 +1410,16 @@ function bbp_get_user_topics_created_url( $user_id = 0 ) {
...
 	}
 
-	// Allow early overriding of the profile URL to cut down on processing
-	$early_url = apply_filters( 'bbp_pre_get_user_topics_created_url', (int) $user_id );
-	if ( is_string( $early_url ) ) {
-		return $early_url;
+	// Bail if intercepted
+	$intercept = bbp_maybe_intercept( 'bbp_pre_get_user_topics_created_url', func_get_args() );
+	if ( bbp_is_intercepted( $intercept ) ) {
+		return $intercept;
```
Comment (mehulkaklotar) [inline src/bp-forums/users/template.php:1416]: @janakkaneriya007 you can revert this as we don't need interception code in the forums. it is used in bbpress's new version to handle filters.

File: src/bp-forums/users/template.php [hunk 2]
```diff
@@ -1475,23 +1468,16 @@ function bbp_get_user_replies_created_url( $user_id = 0 ) {
...
 	}
 
-	// Allow early overriding of the profile URL to cut down on processing
-	$early_url = apply_filters( 'bbp_pre_get_user_replies_created_url', (int) $user_id );
-	if ( is_string( $early_url ) ) {
-		return $early_url;
+	// Bail if intercepted
+	$intercept = bbp_maybe_intercept( 'bbp_pre_get_user_replies_created_url', func_get_args() );
+	if ( bbp_is_intercepted( $intercept ) ) {
+		return $intercept;
```
Comment (mehulkaklotar) [inline src/bp-forums/users/template.php:1474]: @janakkaneriya007 you can revert this as we don't need interception code in the forums. it is used in bbpress's new version to handle filters.

Comment (janakkaneriya007) [reply src/bp-forums/users/template.php]: Fixed!!

File: src/bp-forums/users/template.php (hunk 1 above)
Comment (janakkaneriya007) [reply src/bp-forums/users/template.php]: Fixed!!

**Pull request merged**